src/DropdownReference/components/DropdownReference.ts

```diff
@@ -0,0 +1,120 @@
+import { Component, createElement } from "react";
+import * as classNames from "classnames";
+import Select, { Async, LoadOptionsHandler } from "react-select";
+
+import { Alert } from "../../SharedResources/components/Alert";
+import { Label } from "../../SharedResources/components/Label";
+
+import "react-select/dist/react-select.css";
+import "../../SharedResources/ui/Dropdown.scss";
+
+export interface DropdownReferenceProps {
+    styleObject?: object;
+    labelWidth: number;
+    data: ReferenceOption[];
+    asyncData: LoadOptionsHandler<{}>;
+    value?: string;
+    labelCaption: string;
+    showLabel: boolean;
+    emptyOptionCaption: string;
+    isClearable: boolean;
+    isReadOnly: boolean;
+    selectedValue: ReferenceOption;
+    className?: string;
+    alertMessage: string;
+    searchText: string;
+    loadingText: string;
+    minimumCharacter: number;
+    labelOrientation: "horizontal" | "vertical";
+    readOnlyStyle: "control" | "text";
+    selectType: "normal" | "asynchronous";
+    handleOnchange?: (selectedOption: any) => void;
+}
+
+export interface ReferenceOption {
+    value?: string | boolean;
+    label?: string;
+}
+
+export class DropdownReference extends Component<DropdownReferenceProps> {
+    render() {
+        return this.props.showLabel
+            ? createElement(Label, {
```

Comment:
When the label orientation is "vertical" the component overflows the form width size (-15px margin left and right)

Reply:
Fixed